tools/ci_build/github/azure-pipelines/templates/component-governance-component-detection-steps.yml

```diff
@@ -0,0 +1,15 @@
+# component detection for component governance checks
+parameters:
+- name: condition
+  type: string
+  default: '' # could be 'ci_only', 'always', 'succeeded'
+
+steps:
+- task: ms.vss-governance-buildtask.governance-build-task-component-detection.ComponentGovernanceComponentDetection@0
+  displayName: 'Component Detection'
+  condition:
+    or(or(and(eq('${{parameters.condition}}', 'ci_only'), and(succeeded(), in(variables['Build.Reason'], 'IndividualCI', 'BatchedCI'))),
```

Comment:
I think we could unify it. Just keep one. Then you don't need to put a parameter here. It will be simpler.

Reply:
Was thinking about that, but for the purpose of this change I kept the existing syntax

---
In reply to: [367708662](https://github.com/microsoft/onnxruntime/pull/2857#discussion_r367708662) [](ancestors = 367708662)

Reply:
It has. 
```
in(variables['Build.Reason'], 'Manual ', 'IndividualCI','BatchedCI', 'Schedule').
```

https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml

Reply:
Sorry I meant Build.Reason != CI in test/nuget pipeline

---
In reply to: [367714420](https://github.com/microsoft/onnxruntime/pull/2857#discussion_r367714420) [](ancestors = 367714420)